application/client/src/app/service/session/dependencies/search/state.ts

```diff
@@ -1,26 +1,40 @@
-import { FilterRequest } from './filters/request';
 import { IFilter } from '@platform/types/filter';
 import { Subjects, Subject } from '@platform/env/subscription';
 import { Search } from '@service/session/dependencies/search';
 
 import * as obj from '@platform/env/obj';
 
+export interface IFinish {
+    found: number;
+    error?: string;
+}
+
+export interface ISearchInfo {
+    state: ESearchState;
+    found?: number;
+}
+
+export enum ESearchState {
+    running,
+    finished,
+    dropped,
```

Comment:
value `dropped` isn't used anyhow. Do we really need it? 
I think it would be enough just to have `boolean`: are we searching or not. There are just 2 states. No?